ruma-api-macros/src/api/request.rs

```diff
@@ -85,7 +110,17 @@ impl Request {
                             request,
                         )
                         .into());
-                    }
+                    }},
+                ),
+            };
+
+            quote! {
+                #field_name: match headers
+                    .get(#import_path::exports::http::header::#header_name)
+                    .and_then(|v| v.to_str().ok()) // FIXME: Should have a distinct error message
+                {
+                    Some(header) => #some_case
+                    None => #none_case
```

Comment:
I'd rather have the commas here, rather than in the `some_case` and `none_case` quote invocations above.